### PR TITLE
(indexer) Add tools/near-indexer-framework document with almost everything about the NEAR Indexer Framework

### PR DESCRIPTION
**Pull request by khorolets**
* Move `tools/near-indexer.md` to `tutorials/near-indexer.md`
* Add `tools/near-indexer-framework.md` describing NEAR Indexer Framework
* Update `sidebar.json` a little bit to include those pages
* Add some crosslinks
* Replace the link to NEAR Indexer Framework in `concepts/indexer.md` to point to new document instead of github
 
Basically, we had this entire doc divided into pieces in different places. So @frol and I decided to combine and put in in the docs, so we can refer to it from `README`s

### CODE REVIEW

**Comment by khorolets** [comment]
@thisisjoshford I suggest not to merge it until @frol approves :)

File: docs/tools/near-indexer-framework.md
Comment (frol) [inline docs/tools/near-indexer-framework.md:165]:
```suggestion
As you see NEAR Indexer Framework is a tool, it's not an indexer itself. It is used to create your own indexer. You can find some tutorials about it in the docs:
```

**Review by frol — APPROVED** [review]
We should add a redirect for the moved page

**Comment by khorolets** [comment]
> We should add a redirect for the moved page
@thisisjoshford I'd appreciate your help with this.

**Review by frol — APPROVED** [review]

**Comment by thisisjoshford** [comment]
> > We should add a redirect for the moved page
> 
> @thisisjoshford I'd appreciate your help with this.

absolutely can help redirect this once its merged @khorolets .

**Review by thisisjoshford — CHANGES_REQUESTED** [review]
Looks like CI is failing due to broken links... Can you fix them before merging this @khorolets ?

**Comment by khorolets** [comment]
@thisisjoshford I have tried, but there are no `docs/evm/*` and I've failed to fix that :)

**Review by thisisjoshford — APPROVED** [review]
Seems that there was still a merge conflict with `sidebars.json`. I resolved the conflict and pushed changes @khorolets . You're good to merge when ready.

**Comment by khorolets** [comment]
> absolutely can help redirect this once its merged @khorolets .

I'm merging it, thanks!

**Pull request merged**
